fix: Adding Beta user agreement

alexjba opened the pull request:
### What does the PR do
Closing: #11044 

Changes:
1. Hide AppMain while the fake loading screen is active
2. Create new UserAgreementPopup.qml implementing this design: https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba⎜Desktop?type=design&node-id=31450-560694&t=Q4MOViPCoHsTjhs6-0
3. Add UserAgreementPopup to storybook
4. Show UserAgreementPopup whenever AppMain becomes visible on a production build. I've excluded dev env, not to loose time with this popup on each login and also e2e test environment. Please let me know if you think we should enable it on any of these environments.
<!-- Fill in the relevant information below to help us evaluate your proposed changes. -->

### Affected areas
AppMain
<!-- List the affected areas (e.g wallet, browser, etc..) -->

### Screenshot of functionality (including design for comparison)
<img width="1544" alt="Screenshot 2023-06-14 at 23 50 21" src="https://github.com/status-im/status-desktop/assets/47811206/f72390a2-2c0e-41c4-979f-a7a3666380c4">

<img width="1562" alt="Screenshot 2023-06-14 at 23 59 27" src="https://github.com/status-im/status-desktop/assets/47811206/9aa9317a-6976-4253-8839-d36364791e0c">

- [x] I've checked the design and this PR matches it

<!-- screenshot (or gif/video) that demonstrates the functionality, specially important if it's a bug fix. -->

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 66997aa8 | [#1](https://ci.status.im/job/status-desktop/job/branches/job/linux/job/x86_64/job/tests-imports/job/PR-11100/1/) | 2023-06-14 21:07:01 | ~4 min | `tests/imports` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/branches/job/linux/job/x86_64/job/tests-imports/job/PR-11100/1//consoleText) |

caybro commented on ui/main.qml:
```diff
@@ -322,6 +322,7 @@ StatusWindow {
         id: app
         AppMain {
             sysPalette: systemPalette
+            visible: appLoadingAnimation.active === false
```
```suggestion
            visible: !appLoadingAnimation.active
```

caybro reviewed (approved):
Nice :)

micieslak reviewed (changes requested):
Please add proper sizing to text component (like `Layout.fillWidth: true`) to have proper layout no matter how long text we have.

![image](https://github.com/status-im/status-desktop/assets/20650004/85e7ac00-f779-4793-bc93-6f176b56436d)

Here the text is extended on purpose, similar situation may happen for translations.

alexjba commented:
> Please add proper sizing to text component (like `Layout.fillWidth: true`) to have proper layout no matter how long text we have.
> 
> ![image](https://user-images.githubusercontent.com/20650004/246157147-85e7ac00-f779-4793-bc93-6f176b56436d.png)
> 
> Here the text is extended on purpose, similar situation may happen for translations.
> 
> Btw. things like that can be easily check with the inspection tool (but it's temporarily broken for popups - #11111)

Fixed. I've added wordWrap to the text paragraph and Layout.fillWidth to the checkboxes. Looks good now with large text.

micieslak commented on ui/imports/shared/popups/UserAgreementPopup.qml:
```diff
@@ -0,0 +1,121 @@
+                Layout.preferredWidth: 64
+                Layout.preferredHeight: 64
+                Layout.alignment: Qt.AlignHCenter
+                image.source: Style.png("status-logo")
```
adding `image.mipmap: true` here improves quality a bit bc the original image is much bigger than 64x64

micieslak commented on ui/imports/shared/popups/UserAgreementPopup.qml:
```diff
@@ -0,0 +1,121 @@
+                Layout.fillWidth: true
+                Layout.topMargin: -10 //reduced margin by design
+                Layout.bottomMargin: layout.spacing
+                contentItem: Paragraph {
```
maybe
```
        Layout.fillWidth: true
        Layout.fillHeight: true
```
could be moved from `Paragraph` to `AgreementSection` bc in this context those attached properties make no sense.

micieslak commented on storybook/figma.json:
```diff
@@ -212,5 +212,8 @@
         "https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?type=design&node-id=29528%3A587660&t=GUxMZEWcfRO7pXJb-1"
+    ],
+    "UserAgreementPopup": [
+        "https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba⎜Desktop?type=design&node-id=1023-298171&t=Q4MOViPCoHsTjhs6-0"
```
the link seems to be wrong/outdated

![Screenshot from 2023-06-16 14-30-31](https://github.com/status-im/status-desktop/assets/20650004/f80e21af-75ae-434d-b07c-82cf14537197)

micieslak reviewed (approved):
Nice work! Some minor comments left, but nothing serious.
